samples/apply-blend-renderer-to-hillshade/src/main/java/com/esri/arcgismaps/sample/applyblendrenderertohillshade/components/ApplyBlendRendererToHillshadeViewModel.kt

```diff
@@ -0,0 +1,231 @@
+                app.getString(com.esri.arcgismaps.sample.applyblendrenderertohillshade.R.string.apply_blend_renderer_to_hillshade_app_name)
+    }
+
+    // Raster containing imagery (Shasta).
+    private val imageryRaster: Raster by lazy {
+        Raster.createWithPath("$provisionPath${File.separator}raster-file${File.separator}Shasta.tif")
+    }
+
+    // Raster layer for imagery.
+    private val imageryRasterLayer: RasterLayer by lazy { RasterLayer(imageryRaster) }
+
+    // Raster containing elevation values (grayscale elevation)
+    private val elevationRaster: Raster by lazy {
+        Raster.createWithPath("$provisionPath${File.separator}Shasta_Elevation.tif")
+    }
+
+    // Raster layer for elevation (used as a basemap when applying a color ramp).
+    private val elevationRasterLayer: RasterLayer by lazy { RasterLayer(elevationRaster) }
+
+    // A Basemap for the imagery raster layer.
+    private val imageryBasemap: Basemap by lazy {
+        Basemap(baseLayer = imageryRasterLayer)
+    }
+
+    // A Basemap for the elevation raster layer.
+    private val elevationBasemap: Basemap by lazy {
+        Basemap(baseLayer = elevationRasterLayer)
+    }
+
+    // The ArcGISMap exposed to the UI. The basemap will be
+    // set to the imagery basemap once the raster layers are loaded
+    // and may be switched to the elevation basemap when a color ramp is applied through the UI.
+    val arcGISMap = ArcGISMap()
+
+    // MapViewProxy provided to the MapView composable.
+    val mapViewProxy = MapViewProxy()
+
+    // Message dialog view model to report errors to the UI
+    val messageDialogVM = MessageDialogViewModel()
+
+    // Renderer parameters which can be modified from the UI
+    var altitude by mutableDoubleStateOf(45.0)
+        private set
+
+    var azimuth by mutableDoubleStateOf(0.0)
+        private set
+
+    var slopeType by mutableStateOf<SlopeType?>(SlopeType.None)
+        private set
+
+    // Color ramp presets exposed as user-friendly strings. The underlying ColorRamp is created
+    // on demand in updateRenderer() when a preset is selected.
+    val colorRampPresets = listOf("None", "DEM Light", "Screen Display", "Elevation")
+
+    // Index of the selected color ramp preset; 0 means "None" (no color ramp).
+    var selectedColorRampPresetIndex by mutableStateOf(0)
+        private set
+
+    init {
+        // Load the raster layers and load and set up the map. Perform long-running loads in a coroutine
+        // and surface any errors through the MessageDialogViewModel.
+        viewModelScope.launch {
+            try {
+                // Load both raster layers. If any fails, report the error.
+                elevationBasemap.load().onFailure { throw it }
+                imageryBasemap.load().onFailure { throw it }
+                updateRenderer()
+                arcGISMap.load().onFailure { throw it }
+            } catch (ex: Throwable) {
+               messageDialogVM.showMessageDialog(ex)
```

Comment:
Indentation nitpick: 

```suggestion
                messageDialogVM.showMessageDialog(ex)
```